lib/kafka_rest_client.js

```diff
@@ -169,9 +192,16 @@ KafkaRestClient.prototype.produce = function produce(produceMessage, callback) {
 
 KafkaRestClient.prototype.produceWithRetry = function produceWithRetry(produceMessage, retry, callback) {
     var self = this;
+
     if (self.enable) {
         if (produceMessage.topic in self.cachedTopicToUrlMapping) {
-            var pathUrl = self.cachedTopicToUrlMapping[produceMessage.topic];
+            var pathUrl;
+            if (retry < self.maxRetries - 1) {
+                pathUrl = self.cachedTopicToUrlMapping[produceMessage.topic];
+            } else {
+                pathUrl = self.proxyHost + ':' + self.localAgentPort;
```

Comment:
not very sure about if using self.proxyHost or simply 'localhost', better double-check it.